csrc/host_ir/lowering.cpp

```diff
@@ -84,29 +86,46 @@ std::ostream& operator<<(std::ostream& os, const LoopNest& loop_nest) {
   return os;
 }
 
+int numParallelIterDomains(const TensorView* tv) {
+  return std::ranges::count_if(
+      tv->getLoopDomain(), [](IterDomain* id) { return id->isParallelized(); });
+}
+
+template <typename R>
+TensorView* findMostParallelTensorView(const R& range) {
```

Comment:
You can return a pair of `TensorView, parallel_count` to avoid calling `numParallelIterDomains` again in the caller code below.